sites/frontend/components/Header/Nav/OpenMainMenu/index.js

```diff
@@ -0,0 +1,171 @@
+        width: 8,
+    },
+    ':hover:after': {
+        transform: 'translateY(0) rotate(45deg)',
+    },
+});
+
+type Props = {
+    toggleMainMenu: () => void,
+    showMainMenu: boolean,
+    ariaControls: string,
+};
+
+class OpenMainMenu extends Component<Props, { enhanceCheckbox: boolean }> {
+    constructor(props: Props) {
+        super(props);
+
+        this.state = {
+            enhanceCheckbox: false,
+        };
+    }
+
+    componentDidMount() {
+        /**
+            componentDidMount is only executed in the browser therefore if
+            enhanceCheckbox is set to true it indicates that JS is running 
+            in the browser and we should re-render without the NO JS fallback.
+            Overriding eslint as you can call setState in componentDidMount:
+            https://reactjs.org/docs/react-component.html#componentdidmount
+        * */
+        // eslint-disable-next-line react/no-did-mount-set-state
+        this.setState({
+            enhanceCheckbox: true,
+        });
+    }
+
+    render() {
+        const { toggleMainMenu, ariaControls, showMainMenu } = this.props;
+        const { enhanceCheckbox } = this.state;
+        const CHECKBOX_ID = 'main-menu-toggle';
+
+        if (enhanceCheckbox) {
+            return [
+                <VeggieBurger
+                    showMainMenu={showMainMenu}
+                    toggleMainMenu={toggleMainMenu}
+                    enhanceCheckbox={enhanceCheckbox}
+                    htmlFor={CHECKBOX_ID}
+                    ariaControls={ariaControls}
+                    key="VeggieBurger"
+                />,
+                <OpenMainMenuButton
+                    onClick={() => toggleMainMenu()}
+                    aria-controls={ariaControls}
+                    key="OpenMainMenuButton"
+                >
+                    <ScreenReadable>Show</ScreenReadable>
+                    <OpenMainMenuText>More</OpenMainMenuText>
+                </OpenMainMenuButton>,
+            ];
+        }
+
+        return [
+            <VeggieBurger
+                showMainMenu={showMainMenu}
+                toggleMainMenu={toggleMainMenu}
+                enhanceCheckbox={enhanceCheckbox}
+                htmlFor={CHECKBOX_ID}
+                ariaControls={ariaControls}
+                key="VeggieBurger"
+            />,
+            <OpenMainMenuLabel
+                htmlFor={CHECKBOX_ID}
+                tabindex="0"
+                key="OpenMainMenuLabel"
+            >
+                <ScreenReadable>Show</ScreenReadable>
+                <OpenMainMenuText>More</OpenMainMenuText>
+            </OpenMainMenuLabel>,
+            <OpenMainMenuCheckbox
+                type="checkbox"
+                id={CHECKBOX_ID}
+                aria-controls={ariaControls}
+                tabindex="-1"
+                key="OpenMainMenuCheckbox"
+                role="menuitemcheckbox"
+            />,
+        ];
+    }
+}
+
+export default OpenMainMenu;
```

Comment:
`OpenMainMenu` sounds like a verb rather than a noun. What does this component represent? The Main Menu button / call to action?

Reply:
Yep, `OpenMainMenu` could render as the Main Menu Button OR the Main Menu Checkbox if JS is not running, that's why I didn't call it `MainMenuButton`. What do you think of main `MainMenuToggle`,  "toggle" can be a noun right?

Reply:
Ha, more English language ambiguity! Yeah, I think `Toggle` is reasonable